.test-infra/jenkins/job_beam_PreCommit_Java_MavenInstall.groovy

```diff
@@ -40,6 +40,19 @@ mavenJob('beam_PreCommit_Java_MavenInstall') {
   // Sets that this is a PreCommit job.
   common_job_properties.setPreCommit(delegate, 'Maven clean install')
 
-  // Maven goals for this job.
-  goals('-B -e -Prelease,include-runners,jenkins-precommit,direct-runner,dataflow-runner,spark-runner,flink-runner,apex-runner -DrepoToken=$COVERALLS_REPO_TOKEN -DpullRequest=$ghprbPullId help:effective-settings clean install coveralls:report')
+  // Maven goals for this job: The Java SDK, its dependencies, and things that depend on it.
+  goals('''\
+    --batch-mode \
+    --errors \
+    --activate-profiles release,jenkins-precommit,direct-runner,dataflow-runner,spark-runner,flink-runner,apex-runner \
+    --projects sdks/java/core \
```

Comment:
I believe there exist modules that will be written in java and part of our repository that will not depend on the core SDK; I'm not sure how to meaningfully catch them, though

Reply:
I don't have a great plan. I expect things to evolve from this crude step.

Reply:
Yeah, this is more important for the build pipeline stuff